src/index.js

```diff
@@ -249,6 +259,18 @@ const MicroModal = (() => {
       options.triggers = [...value]
       new Modal(options) // eslint-disable-line no-new
     }
+
+    // Watch document for esc key to close modals top to bottom
+    document.addEventListener('keydown', onKeyDown)
+  }
+
+  const onKeyDown = event => {
```

Comment:
The new `onKeyDown` event handler ensures that the esc key will open close the top-most modal